models/articles-models.js

```diff
@@ -21,22 +21,43 @@ exports.selectArticles = (sort_by = "created_at", order = "desc") => {
     "comment_count",
   ];
   const validOrder = ["asc", "desc"];
+  const validTopic = []
+  const queryValues = []
+  let topicString = `SELECT topic FROM articles`
 
   if (!validSortBy.includes(sort_by) || !validOrder.includes(lowerCaseOrder)) {
     return Promise.reject({ status: 400, message: "Bad Request." });
   }
 
-  let queryString = format(`SELECT articles.author, articles.title, articles.topic, articles.created_at, articles.votes, articles.article_img_url, 
-      COUNT(comments.article_id) 
-      AS comment_count 
+  return db.query(topicString).then((result)=>{
+    result.rows.forEach((row) => {
+      validTopic.push(row.topic)
+    })
```

Comment:
This works, but it is not very clear what you're trying to do. Rather than predefining and pushing to the validTopic array, why not just return the rows from this .then into the next one, on a name like `validTopics` ?